src/client/components/equalizer/equalizer.test.js

```diff
@@ -0,0 +1,444 @@
+      expect(wrapper).toMatchSnapshot();
+    });
+  });
+
+  describe('test equlizer when sound is uploaded', () => {
+    const initialState = {
+      audioData: {
+        analyser: {},
+        audioContext: {},
+        widthCanvas: 100,
+        heightCanvas: 200,
+        sound: true,
+      },
+    };
+    let wrapper;
+    let instance;
+    beforeEach(() => {
+      wrapper = setup(initialState);
+      instance = wrapper.instance();
+    });
+
+    it('should render `play` button', () => {
+      expect(wrapper.find('Button[value="Play"]').length).toBe(1);
+    });
+
+    it('should render `pause` button', () => {
+      expect(wrapper.find('Button[value="Pause"]').length).toBe(1);
+    });
+
+    it('should call function after click on `play` button', () => {
+      const spyPlay = jest.spyOn(instance, 'playSoundFromFile');
+
+      wrapper.instance().forceUpdate();
+      wrapper.find('Button[value="Play"]').simulate('click');
+
+      expect(spyPlay).toHaveBeenCalled();
+    });
+
+    it('should call function after click on `pause` button', () => {
+      const spyPause = jest.spyOn(instance, 'pauseSoundFromFile');
+
+      wrapper.instance().forceUpdate();
+      wrapper.find('Button[value="Pause"]').simulate('click');
+
+      expect(spyPause).toHaveBeenCalled();
+    });
+
+    it('should call function after click on `startStreamButton` button', () => {
+      const spyPause = jest.spyOn(instance, 'startMuteStream');
+
+      wrapper.instance().forceUpdate();
+      wrapper.find('Button[value="Start stream"]').simulate('click');
+
+      expect(spyPause).toHaveBeenCalled();
+    });
+  });
+
+  describe('test component methods', () => {
+    const initialState = {
+      audioData: {
+        analyser: {
+          getByteFrequencyData: jest.fn(),
+        },
+        audioContext: {},
+        widthCanvas: 100,
+        heightCanvas: 200,
+        cxt: {},
+        sound: {
+          pause: () => (
+            'fakePauseMethod'
+          ),
+          play: () => (
+            'fakePlayMethod'
+          ),
+          stop: () => 'fakePauseMethod',
+        },
+        voice: {
+          play: () => 'fakePlayMethod',
+          stop: () => 'fakeStopMethod',
+          pause: () => 'fakePauseMethod',
+          connect: () => 'fakeConnectMethod',
+        },
+        startMuteState: false,
+        playPauseState: false,
+      },
+    };
+
+    let wrapper;
+    let instance;
+    beforeEach(() => {
+      jest.resetAllMocks();
+      wrapper = setup(initialState);
+      instance = wrapper.instance();
+    });
+
+    it('will test own method `setCanvasToState` which should set in state reference on canvas element', () => {
+      const paramForTest = document.createElement('canvas');
+      const testedFunction = wrapper.instance().setCanvasToState;
+
+      paramForTest.getContext = jest.fn(() => 'fakeCanvas');
+
+      testedFunction(paramForTest);
+      expect(wrapper.instance().state.ctx).toEqual('fakeCanvas');
```

Comment:
You already have declared wrapper.instance() in upper scope, why don’t you use it here?

Reply:
used in all test - missed there. Done